tests/icons_entry.py

```diff
@@ -1,38 +1,38 @@
 #!/usr/bin/env python3
+
 """
-# Copyright (C) 2016
-# This program is free software: you can redistribute it and/or modify
-# it under the terms of the GNU General Public License (version 3+) as
-# published by the Free Software Foundation. You should have received
-# a copy of the GNU General Public License along with this program.
-# If not, see <http://www.gnu.org/licenses/>.
+Copyright (C) 2019 Numix Project
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License (version 3+) as
+published by the Free Software Foundation. You should have received
+a copy of the GNU General Public License along with this program.
+If not, see <http://www.gnu.org/licenses/>.
 """
+
 from glob import glob
+from json import load
 from os import path
-import json
 
-from utils import error
+from utils import error, THEMES
 
 ABS_PATH = path.dirname(path.abspath(__file__))
 DB_FILE = path.join(ABS_PATH, "../data.json")
-THEMES = ["circle", "square"]
 ICONS_DIR = path.join(ABS_PATH, "../icons/")
 
-
-has_errors = False
 with open(DB_FILE, 'r') as db_obj:
-    entries = json.load(db_obj).keys()
+    entries = load(db_obj).keys()
 
+has_errors = False
 reported = []
 for theme in THEMES:
     icons = glob(path.join(ICONS_DIR, theme, "48") + "/*.svg")
     for icon in icons:
         icon_name = path.splitext(path.basename(icon))[0]
-        if icon_name not in entries and icon_name not in reported:
-            reported.append(icon_name)
-            has_errors = True
-            error("The icon {} doesn't have any "
-                  "entry in the database.".format(
-                      icon_name
-                  ))
-exit(int(has_errors))
+        if icon_name in entries or icon_name in reported:
```

Comment:
As well as the previous comment about `continue` being cleaner than a whole new `if` level, this has an efficiency improvement. For those who know logic, you'll recognise that:

```
(¬A) ʌ (¬B) === ¬(A v B)
```

Since the continue handles the `NOT` on the right hand side, we're actually checking whether just `(A v B)`. Since `A` and `B` are just inclusion events (`x in X`) they're much quicker than evaluating the previous exclusion events (`x not in X`). This is further improved because Python halts logical evaluations as soon as it knows the statement as a whole is true, i.e. here if `icon_name in entries == True` it doesn't bother evaluating `icon_name in reported` and continues to the `continue`.